File: frontend/src/api/samplingQuality.ts
import { apiClient } from '@/api/client';

export interface SamplingCriterion {
  name: string;
  score: number;
  max: number;
  detail: string;
  recommendation: string;
}

export interface SamplingQuality {
  diagnostic_id: string;
  overall_score: number;
  grade: string;
  criteria: SamplingCriterion[];
  confidence_level: string;
  warnings: string[];
  evaluated_at: string;
}

export interface BuildingSamplingQuality {
  building_id: string;
  avg_score: number;
  worst_diagnostic: string | null;
  best_diagnostic: string | null;
  diagnostics: SamplingQuality[];
  evaluated_at: string;
}

export const samplingQualityApi = {
  getDiagnostic: async (diagnosticId: string): Promise<SamplingQuality> => {
    const response = await apiClient.get<SamplingQuality>(
      `/diagnostics/${diagnosticId}/sampling-quality`,
    );
    return response.data;
  },

  getBuilding: async (buildingId: string): Promise<BuildingSamplingQuality> => {
    const response = await apiClient.get<BuildingSamplingQuality>(
      `/buildings/${buildingId}/sampling-quality`,
    );
    return response.data;
  },
};

File: backend/app/schemas/sampling_quality.py
"""Pydantic v2 schemas for Sampling Quality Score."""

from pydantic import BaseModel, ConfigDict


class SamplingCriterionRead(BaseModel):
    """Single criterion result."""

    name: str
    score: int
    max: int
    detail: str
    recommendation: str

    model_config = ConfigDict(from_attributes=True)


class SamplingQualityRead(BaseModel):
    """Per-diagnostic sampling quality score."""

    diagnostic_id: str
    overall_score: int
    grade: str  # A, B, C, D, F
    criteria: list[SamplingCriterionRead]
    confidence_level: str  # high, medium, low, very_low
    warnings: list[str]
    evaluated_at: str

    model_config = ConfigDict(from_attributes=True)


class BuildingSamplingQualityRead(BaseModel):
    """Building-level aggregate of sampling quality across diagnostics."""

    building_id: str
    avg_score: int
    worst_diagnostic: str | None
    best_diagnostic: str | None
    diagnostics: list[SamplingQualityRead]
    evaluated_at: str

    model_config = ConfigDict(from_attributes=True)
